packages/aws-cdk-lib/aws-cloudfront-origins/test/s3-bucket-origin.test.ts

```diff
@@ -467,4 +467,404 @@ describe('S3BucketOrigin', () => {
+                          ],
+                        ],
+                      },
+                    },
+                  },
+                ],
+              },
+            },
+          },
+          MyDistributionB4B294FCF: {
+            Type: 'AWS::CloudFront::Distribution',
+            Properties: {
+              DistributionConfig: {
+                DefaultCacheBehavior: {
+                  CachePolicyId: '658327ea-f89d-4fab-a63d-7e88639e58f6',
+                  Compress: true,
+                  TargetOriginId: 'MyDistributionBOrigin12F11A471',
+                  ViewerProtocolPolicy: 'allow-all',
+                },
+                Enabled: true,
+                HttpVersion: 'http2',
+                IPV6Enabled: true,
+                Origins: [
+                  {
+                    DomainName: {
+                      'Fn::GetAtt': [
+                        'MyBucketF68F3FF0',
+                        'RegionalDomainName',
+                      ],
+                    },
+                    Id: 'MyDistributionBOrigin12F11A471',
+                    S3OriginConfig: {
+                      OriginAccessIdentity: {
+                        'Fn::Join': [
+                          '',
+                          [
+                            'origin-access-identity/cloudfront/',
+                            {
+                              Ref: 'MyDistributionAOrigin1S3Origin8C75F420',
+                            },
+                          ],
+                        ],
+                      },
+                    },
+                  },
+                ],
+              },
+            },
+          },
+        });
+      });
+    });
+
+    describe('when using an imported bucket from another stack', () => {
+      let distributionStack: Stack;
+      let bucketStack: Stack;
+      let bucket: s3.IBucket;
+      let origin: cloudfront.IOrigin;
+
+      beforeAll(() => {
+        const app = new App();
+        distributionStack = new Stack(app, 'distributionStack');
+        bucketStack = new Stack(app, 'bucketStack');
+        bucket = s3.Bucket.fromBucketName(bucketStack, 'BucketId', 'my-bucket');
+        origin = origins.S3BucketOrigin.withOriginAccessIdentity(bucket);
+
+        new cloudfront.Distribution(distributionStack, 'MyDistributionA', {
+          defaultBehavior: { origin: origin },
+        });
+      });
+
+      it('should warn user bucket policy is not updated', () => {
+        Annotations.fromStack(distributionStack).hasWarning('/distributionStack/MyDistributionA/Origin1',
+          'Cannot update bucket policy of an imported bucket. You may need to update the policy manually instead.\n' +
+          'See the "Using pre-existing S3 buckets" section of module\'s README for more info. [ack: @aws-cdk/aws-cloudfront-origins:updateImportedBucketPolicy]');
+      });
+
+      it('should create OAI in bucket stack and output it, then reference the output in the distribution stack', () => {
```

Comment:
This is specifically for testing the cross stack logic in the OAI bind method: https://github.com/samson-keung/aws-cdk/blob/9f5c1430bec9dd41246b83085cd4350b1ab1dffb/packages/aws-cdk-lib/aws-cloudfront-origins/lib/s3-bucket-origin.ts#L195-L199

Such logic is not in `withOriginAccessControl`, do we need it there?

Reply:
We won't be able to handle cross-stack references as easily for OAC because the bucket policy references the distribution, so the solution is to create the bucket and distribution in the same stack. Versus OAI where it just references the OAI user (so creating the OAI in the bucket stack easily resolves the circular dependency).